SpatialGDK/Source/SpatialGDKTests/Private/LocalDeploymentManager/LocalDeploymentManagerTest.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "LocalDeploymentManager.h"
+#include "SpatialGDKDefaultLaunchConfigGenerator.h"
+#include "SpatialGDKDefaultWorkerJsonGenerator.h"
+#include "SpatialGDKEditorSettings.h"
+
+#include "CoreMinimal.h"
+
+#define LOCALDEPLOYMENT_TEST(TestName) \
+	TEST(Services, LocalDeployment, TestName)
+
+namespace
+{
+	// TODO: UNR-1969 - Prepare LocalDeployment in CI pipeline
+	const double MAX_WAIT_TIME_FOR_LOCAL_DEPLOYMENT_OPERATION = 10.0;
+
+	// TODO: UNR-1964 - Move EDeploymentState enum to LocalDeploymentManager
+	enum class EDeploymentState { IsRunning, IsNotRunning };
+
+	const FName AutomationWorkerType = TEXT("AutomationWorker");
+	const FString AutomationLaunchConfig = TEXT("Improbable/AutomationLaunchConfig.json");
+
+	FLocalDeploymentManager* GetLocalDeploymentManager()
+	{
+		FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+		FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+		return LocalDeploymentManager;
+	}
+
+	bool GenerateWorkerAssemblies()
+	{
+		FString BuildConfigArgs = TEXT("worker build build-config");
+		FString WorkerBuildConfigResult;
+		int32 ExitCode;
+		const FString SpatialExe(TEXT("spatial.exe"));
+		FSpatialGDKServicesModule::ExecuteAndReadOutput(SpatialExe, BuildConfigArgs, FSpatialGDKServicesModule::GetSpatialOSDirectory(), WorkerBuildConfigResult, ExitCode);
+
+		const int32 ExitCodeSuccess = 0;
+		return (ExitCode == ExitCodeSuccess);
+	}
+
+	bool GenerateWorkerJson()
+	{
+		const FString WorkerJsonDir = FSpatialGDKServicesModule::GetSpatialOSDirectory(TEXT("workers/unreal"));
+
+		FString JsonPath = FPaths::Combine(WorkerJsonDir, TEXT("spatialos.UnrealAutomation.worker.json"));
+		if (!FPaths::FileExists(JsonPath))
+		{
+			bool bRedeployRequired = false;
```

Comment:
```suggestion
```

Reply:
Unfortunately that's a mandatory outArgument that's not used here.

Reply:
Don't mind me